ansible_collections/arista/avd/plugins/modules/eos_validate_state_reports.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+
+# NOTE: This is supposed to be deprecated as per
+# https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#ansible-metadata-block
+# But our doc Jinja2 template renders it as preview which is what we want
+ANSIBLE_METADATA = {"metadata_version": "1.0", "status": ["preview"]}
+
+DOCUMENTATION = r"""
+---
+module: eos_validate_state_reports
+version_added: "4.5.0"
+author: Arista Ansible Team (@aristanetworks)
+short_description: Generates validation reports for the eos_validate_state role
+description:
+  - The C(arista.avd.eos_validate_state_reports) module, an Ansible Action Plugin, is designed to generate
+    validation reports from the test results produced by the ANTA test framework.
+  - This plugin requires a temporary JSON file for each host in the Ansible play, which contains all test results.
+    The path to this temporary JSON file, used by the plugin, is supplied in the hostvars by the `eos_validate_state_runner` plugin.
+  - |-
+    The plugin offers the following functionalities:
+      - It aggregates all test results from every host in the Ansible play and generates a CSV report.
+      - It produces a detailed Markdown report with various sections presenting key statistics derived from the results.
+      - The design allows for easy extension to support additional report formats in the future.
+options:
+  validation_report_csv:
+    description: Indicates whether a CSV report should be generated.
+    type: bool
+    default: true
+  validation_report_md:
+    description: Indicates whether a Markdown report should be generated.
+    type: bool
+    default: true
+  only_failed_tests:
+    description: Determines if the generated reports should include only the failed tests.
+    type: bool
+    default: false
+  csv_report_path:
+    description:
+      - The absolute path where the CSV report will be saved.
+      - Required if C(validation_report_csv) is set to C(True).
+    type: str
+  md_report_path:
+    description:
+      - The absolute path where the Markdown report will be saved.
+      - Required if C(validation_report_md) is set to C(True).
+    type: str
```

Comment:
We are using these variables today in eos_validate_state:
https://avd.sh/en/stable/roles/eos_validate_state/index.html#default-variables

Since we provide a default path for the CSV and MD reports, I think it makes sense to just have a boolean to turn on/off the report generation.

Reply:
understood - this makes for a lots of knobs but I guess people can deal with it